### ISSUE TITLE
No mismatches shown from local BAM file?

### ISSUE THREAD

**Issue by ChristianRohde** [gladkia/igvShiny#47]
Hi, 

I am using your example to show bam/cram files from WES data. I wonder if it is a bug that the locally loaded bam file does not show any mismatches:

![Screenshot 2023-07-17 at 16 38 48](https://github.com/paul-shannon/igvShiny/assets/63801549/10c5b8c3-9bdc-4327-aaec-1b45ca932409)

I already implemented this function to load our mapped WES data, which we stored as BAM file. Using the ScanBamParam(which=region.GRanges) I can now efficiently load data from rather huge bam files. 

![Screenshot 2023-07-17 at 17 05 04](https://github.com/paul-shannon/igvShiny/assets/63801549/167ccc55-0ceb-49f3-8eef-26c2379c0229)

Unfortunately, also here I do not find any mismatch but only Insertions highlighted. Is this a bug in some igvshiny function or rather a feature of the local bam files including mine? Do I have to modify my bam file to visualize mismatches ? 

Best,
Christian

**Comment by paul-shannon** [comment]
Hi Christian,

A few weeks ago I retired from the Institute for Systems Biology.  I’m afraid I don’t have time to look into this.  
(I should formalize and publicize my retirement from igvShiny and igvR.)

One thing you might try, since Vince has expressed a little interest in taking on the packages, is to submit an issue here:

   https://github.com/vjcitn/igvShiny

That might go over best if you included a couple of things:
- a report on whether the problem occurs in the latest igv.js, Jim Robinson’s foundational javascript library, which both igvShiny and igvR use?
- maybe even a PR with the fix, if you have the time and capacity to figure it out

Sorry I cannot be more helpful.

- Paul

**Comment by ChristianRohde** [comment]
Hi Paul,

this sounds good. Thank you for your enormous support in the past and also the good hints here. For sure I will try to follow up on this, since igvshiny is a very helpful tool. 

Best wishes,
Christian

**Comment by ChristianRohde** [comment]
Maybe I closed this to early: it seems that I still do not understand GitHub enough. At the moment there is no way to contact @vjcitn opening an issue: maybe this is set by purpose, I am not sure. If there is any information how igvshiny hopefully is kept alive in future please give me some hints.

**Comment by gladkia** [comment]
Hi, @ChristianRohde thanks for the detailed info on how to make `loadBamTrackFromLocalData`. Do you think that some additional actions are required for this issue? Can the issue be closed?

**Comment by ChristianRohde** [comment]
Hi @gladkia, 

thank you for taking care. I really like igvShiny and use it a lot to visualize our data. 

Right for me the case is closed and I added all information which are necessary to import reads from local BAM files including the actual sequence information. I can imagine that it might make sense to add my information to another Demo or at least update the Demo I used to understand the workflow. In line 149 we have of https://github.com/gladkia/igvShiny/blob/master/inst/demos/igvShinyDemo.R we have

```r
x <- readGAlignments(bamFile)
```  
I checked my code: it is enough that you add the following parameter to load and visualize the sequence.

```r
x <- readGAlignments(bamFile, param = ScanBamParam(what="seq"))
```  
On the other hand the question how to import just the part of the BAM we inspect is crucial as well if it comes to real data. But this might better be part of another Demo to keep this one simple?

Best,
Christian
